projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache_Setup.php

```diff
@@ -26,10 +27,12 @@ public static function run_setup() {
 			$result = self::$step();
 
 			if ( $result instanceof Boost_Cache_Error ) {
+				Analytics::record_user_event( 'cache_setup_error', array( 'error_message' => $result->get_error_code() ) );
 				return $result->to_wp_error();
 			}
 
 			if ( is_wp_error( $result ) ) {
+				Analytics::record_user_event( 'cache_setup_error', array( 'error_message' => $result->get_error_code() ) );
```

Comment:
I just saw there's a boost_page_cache_setup_failed in run_setup.php which is similar to cache_setup_error. We possibly don't need this one, or more likely the other one as run_setup() is called in several places.

Reply:
I would have said the opposite because run_setup is called from two places but only tracked in one. Your tracks made sure they were tracked in both places!

Reply:
I didn't realize that until you pointed it out - thanks!

I've moved both success and failed tracking events to the `Page_Cache_Setup::run_setup` instead of the DS callback.

I've also updated the names of the JS events, so they share the same structure (start with `page_cache_` at least).